### PR TITLE
ci: auto fix default_deferrable value with LibCST

### PR DESCRIPTION
**Pull request by Lee-W**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by uranusjr — APPROVED** [review]

**Comment by bugraoz93** [comment]
Hello @Lee-W, I have got an error about the hook introduced in this PR not being able to be installed via `Python 3.8`. I think it's due to `"libcst>=1.4.0"`. 
I know `Python 3.8` is nearly EOL, so in Airflow 3, the decision was to delete EOL Python versions. Until the Airflow 3 release date, `Python 3.8` will lose support.
Does this mean we should start developing at least in `Python 3.9`? 

>Which Python version should we support in Airflow 3?
    The consensus was to keep things as-is i.e following the current [Python-support policy](https://github.com/apache/airflow?tab=readme-ov-file#support-for-python-and-kubernetes-versions) where we would remove the support for a Python version when it is EOL.

**Comment by Lee-W** [comment]
it's something I missed. just created a pr for the fix https://github.com/apache/airflow/pull/42083

**Comment by bugraoz93** [comment]
Thanks, @Lee-W!

File: scripts/ci/pre_commit/check_deferrable_default.py [hunk 1]
```diff
@@ -33,62 +37,71 @@
...
-        and isinstance(default.args[0], ast.Constant)
-        and default.args[0].value == "operators"
-        and isinstance(default.args[1], ast.Constant)
-        and default.args[1].value == "default_deferrable"
-        and len(default.keywords) == 1
-        and default.keywords[0].arg == "fallback"
-        and isinstance(default.keywords[0].value, ast.Constant)
-        and default.keywords[0].value.value is False
-    )
+    return ast.unparse(default) == "conf.getboolean('operators', 'default_deferrable', fallback=False)"
```
Comment (ashb) [inline scripts/ci/pre_commit/check_deferrable_default.py:81]: This also needs Python 3.9 to work, it didn't exist before there

**Pull request merged**